.github/labeler.yml

```
@@ -0,0 +1,73 @@
+area:appsec:
+- tracer/src/Datadog.Trace/AppSec/**/*
+- tracer/test/Datadog.Trace.Security.*/**/*
+
+area:benchmarks:
+- tracer/test/benchmarks/**/*
+- tracer/src/Datadog.Trace.BenchmarkDotNet/**/*
+
+area:builds:
+- tracer/build/**/*
+
+area:ci-app:
+- tracer/src/Datadog.Trace/Ci/**/*
+- tracer/src/Datadog.Trace.MSBuild/**/*
+
+area:customer-samples:
+- tracer/samples/**/*
+
+area:dependabot:
+- tracer/dependabot/**/*
+
+area:docs:
+- docs/**/*
+- blog/**/*
+- '**/README*'
+- '**/*.md'
+
+area:installers:
+- tracer/src/WindowsInstaller/**/*
+
+area:integrations:
+- tracer/src/Datadog.Trace/ClrProfiler/**/*
+- tracer/src/Datadog.Trace.AspNet/**/*
+- tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/**/*
+- tracer/test/Datadog.Trace.ClrProfiler.Managed.Tests/**/*
+
+area:native-library:
+- tracer/src/Datadog.Trace.ClrProfiler.Native/**/*
+- tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/**/*
+- tracer/src/Datadog.AutoInstrumentation.NativeLoader/**/*
+- tracer/test/Datadog.Trace.ClrProfiler.Native.Tests/**/*
+- tracer/test/Datadog.AutoInstrumentation.NativeLoader.Tests/**/*
+
+area:opentracing:
+- tracer/src/Datadog.Trace.OpenTracing/**/*
+- tracer/test/Datadog.Trace.OpenTracing.*/**/*
+
+area:shared-components:
+- shared/**/*
+
+area:tests:
+- test/**/*
+- tracer/test/**/*
+
+area:test-apps:
+- tracer/test/test-applications/**/*
+
+area:tools:
+- tracer/src/Datadog.Trace.Tools.*/**/*
+- tracer/test/Datadog.Trace.Tools.Analyzers.Tests/**/*
+
+area:tracer:
+# This should be: if any modified file is in tracer/src/Datadog.Trace, excluding files in tracer/src/Datadog.Trace/ClrProfiler.
+# However, the following line makes the action fail with "Error: glob pattern string required"
+# even though it _should_ work:
+# - any: ['tracer/src/Datadog.Trace/**/*', '!tracer/src/Datadog.Trace/ClrProfiler/**/*']
+# so don't exclude `tracer/src/Datadog.Trace/ClrProfiler` for now
+- tracer/src/Datadog.Trace/**/*
+- tracer/test/Datadog.Trace.Tests/**/*
+- tracer/test/Datadog.Trace.IntegrationTests/**/*
```

I don't think your failing pattern was correct - I think you should try this:

```suggestion
area:tracer:
- any: ['tracer/src/Datadog.Trace/**/*', 'tracer/test/Datadog.Trace.Tests/**/*', 'tracer/test/Datadog.Trace.IntegrationTests/**/*']
- all: ['!tracer/src/Datadog.Trace/ClrProfiler/**/*']
```

---

They have an example of this in https://github.com/actions/labeler#common-examples.
```
# Add 'source' label to any change to src files within the source dir EXCEPT for the docs sub-folder
source:
- any: ['src/**/*', '!src/docs/*']
```

By using `all`, then the `area:tracer` will only be set if none of the changed files are in `tracer/src/Datadog.Trace/ClrProfiler/`. This matters if you have a PR with changed files in both  `tracer/src/Datadog.Trace/` and `tracer/src/Datadog.Trace/ClrProfiler/` because we would want both `area:tracer` and `area:integrations` labels.

---

I accepted your suggestion, though. I think we missing label is better than a wrong label.

---

Reverting the suggestion. Now if any modified file is not in `tracer/src/Datadog.Trace/ClrProfiler/`, it matches the `all` clause and applies the `area:tracer` tag.

---

Ahh. I figured it out. The template uses `actions/labeler@v2`. v3 was released months ago and supports the syntax we want.